src/home/components/PetWeightPage/index.js

```diff
@@ -0,0 +1,160 @@
+import UploadImage from '../../../components/UploadImage';
+import RangeSlider from '../../../components/RangeSlider';
+import TextInput from '../../../components/TextInput';
+import RadioButton from '../../../components/RadioButton';
+import Button from '../../../components/Button';
+import './index.scss';
+
+const html = `
+  <div class="pet-weight-page">
+    <div class="pet-weight-page__content" data-select="container">
+      <div class="pet-weight-page__image-container" data-select="image-container"></div>
+
+      <div class="pet-weight-page__description">
+        <h1 class="pet-weight-page__title">Qual é o peso do seu animal de estimação?</h1>
+        <p class="pet-weight-page__hint">Ajuste de acordo com a realidade</p>
+      </div>
+
+      <div class="pet-weight-page__slider-container" data-select="slider-container"></div>
+
+      <div class="pet-weight-page__inputs" data-select="input-container">
+      </div>    
+    </div>
+  </div>;
+`;
+
+export default function PetWeightPage({ petPhoto }) {
+  Component.call(this, { html });
+  this.initializeComponents();
+  this.setupEventListeners();
+  this.styleComponents();
+  this.petPhoto = petPhoto;
+}
+
+PetWeightPage.prototype = Object.assign(
+  PetWeightPage.prototype,
+  Component.prototype,
+  {
+    initializeComponents() {
+      const $container = this.selected.get('container');
+      const $imageContainer = this.selected.get('image-container');
+      const $sliderContainer = this.selected.get('slider-container');
+      const $inputsContainer = this.selected.get('input-container');
+
+      this.setupComponents(
+        $container,
+        $imageContainer,
+        $sliderContainer,
+        $inputsContainer,
+      );
+    },
+
+    setupComponents(
+      $container,
+      $imageContainer,
+      $sliderContainer,
+      $inputsContainer,
+    ) {
+      this.image = new UploadImage();
+      this.slider = new RangeSlider();
+      this.input = new TextInput({
+        placeholder: 'Peso',
+        assetUrl: '',
+        assetPosition: 'prefix',
+        variation: 'standard',
+      });
+      this.radioKG = new RadioButton({
+        check: true,
+        text: 'KG',
+        value: 'kg',
+        name: 'weight-unit',
+      });
+      this.radioLB = new RadioButton({
+        text: 'LB',
+        value: 'lb',
+        name: 'weight-unit',
+      });
+      this.button = new Button({
+        text: 'Continuar',
+        isFullWidth: false,
+        isDisabled: false,
+      });
+
+      // Mount components
+      this.image.mount($imageContainer);
+      this.slider.mount($sliderContainer);
+      this.input.mount($inputsContainer);
+      this.radioKG.mount($inputsContainer);
+      this.radioLB.mount($inputsContainer);
+      this.button.mount($container);
+    },
+
+    styleComponents() {
+      // Add classes to components
+      this.addClassToComponent(
+        this.image.selected.get('image-preview'),
+        'pet-weight-page__image',
```

Comment:
porque não apenas `this.image.selected.get('image-preview').classList.add('pet-weight-page__image')`? Entendo o problema que tentou resolver, mas acho que deu meio que na mesma, ainda estamos escrevendo bastante, acredito que não precisamos dessa função e pode colocar uma declaração em baixo da outra sem nada muito fancy